command/connect/envoy/exec_test.go

```diff
@@ -0,0 +1,178 @@
+// +build linux darwin
+
+package envoy
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"reflect"
+	"testing"
+	"time"
+)
+
+func TestExecEnvoy(t *testing.T) {
+	cmd, destroy := helperProcess("exec-fake-envoy")
+	defer destroy()
+
+	cmd.Stderr = os.Stderr
+	outBytes, err := cmd.Output()
+	if err != nil {
+		t.Fatalf("error launching child process: %v", err)
+	}
+
+	var got FakeEnvoyExecData
+	if err := json.Unmarshal(outBytes, &got); err != nil {
+		t.Fatalf("error decoding response from child: %v", err)
+	}
+
+	expectArgs := []string{
+		"--v2-config-only",
+		"--disable-hot-restart",
+		"--config-path",
+		"/dev/fd/3",
+		"--fake-envoy-arg",
+	}
+	expectConfigPath := "/dev/fd/3"
+	expectConfigData := fakeEnvoyTestData
+
+	if !reflect.DeepEqual(expectArgs, got.Args) {
+		t.Fatalf("expected args to be %v but instead they were %v", expectArgs, got.Args)
+	}
+	if expectConfigPath != got.ConfigPath {
+		t.Fatalf("expected config path to be %q but instead it was %q", expectConfigPath, got.ConfigPath)
+	}
+	if expectConfigData != got.ConfigData {
+		t.Fatalf("expected config data to be %q but instead it was %q", expectConfigData, got.ConfigPath)
+	}
```

Comment:
We prefer to use `testify/require` and optionally `testify/assert` for test assertions because they are terser and have more useful and consistent output on failure. `require` and `assert` have an identical interface, require fails the test immediately on a failure so is important for error/nil checks where continuing might cause a panic due to use of a nil response or similar. We often only use require as it's simpler and almost always needed for a error check during setup at least, but in some more involved tests with lots of independent assertions you might want to use `assert` for those just so you can see multiple failures at once.

The 4 assertions in this test would look like:

```
require := require.New(t) // somewhere near top of test func for convenience

require.NoError(json.Unmarshal(outBytes, &got))

require.Equal(expectArgs, got.Args)
require.Equal(expectConfigPath, got.ConfigPath)
require.Equal(expectConfigData, got.ConfigData)
```

Note that Equal will do a DeepEqual equivalent and on error outputs a useful message including the values but also a diff of just the changed fields in case of a deep struct/map.